go/vt/wrangler/schema.go

```diff
@@ -227,6 +233,25 @@ func (wr *Wrangler) ValidateSchemaKeyspace(ctx context.Context, keyspace string,
 	er := concurrency.AllErrorRecorder{}
 	wg := sync.WaitGroup{}
 
+	// If we are checking against the vschema then all shards
+	// should just be validated individually against it
+	if includeVSchema {
+		for _, shard := range shards {
+			wg.Add(1)
+			err := wr.validateSchemaShardConcurrent(ctx, keyspace, shard, excludeTables, includeViews, includeVSchema, &wg)
```

Comment:
the vschema object exists at the keyspace, not shard, level. i think this isn't quite the refactor we want.

i think it should be more like:

```go
func (wr *Wrangler) ValidateSchemaKeyspace(ctx, ....) {
    // other stuff
    if includeVSchema {
        wr.validateVSchema(ctx, keyspace)
}

func (wr *Wrangler) ValidateSchemaShard(ctx ....., includeVSchema bool) {
    // other stuff
    if includeVSchema {
        wr.validateVSchema(ctx, keyspace)
    }
}
```

where `validateVSchema` is this [this bit](https://github.com/vitessio/vitess/pull/8012/files#diff-325df117ec2cfabd09085e4914cd447d075be627b12d7aec1fcb74a1b038bda5R169-R183)